components/match2/commons/match_ticker/match_ticker_display.lua

```diff
@@ -0,0 +1,325 @@
+	return self
+end
+
+function Match:upperRow(upperRow)
+	self.upperRow = upperRow
+	return self
+end
+
+function Match:lowerRow(lowerRow)
+	self.lowerRow = lowerRow
+	return self
+end
+
+function Match:create()
+	return self.root
+		:node(self.upperRow)
+		:node(self.lowerRow)
+end
+
+local UpperRow = Class.new(
+	function(self)
+		self.root = mw.html.create('tr')
+	end
+)
+
+function UpperRow:addOpponent(opponent, side, noLink)
+	opponent = MatchTickerDisplay.Opponent.fromMatch2Record(opponent)
+	local OpponentDisplay
+
+	-- catch empty and 'TBD' opponents
+	if HelperFunctions.opponentIsTbdOrEmpty(opponent) then
+		OpponentDisplay = mw.html.create('i')
+			:wikitext(_TBD)
+	else
+		OpponentDisplay = MatchTickerDisplay.OpponentDisplay.InlineOpponent{
+			opponent = opponent,
+			teamStyle = 'short',
+			flip = side == _LEFT_SIDE,
+			showLink = not noLink
+		}
+	end
+
+	self[side] = mw.html.create('td')
+		:addClass('team-' .. (_SIDE_CLASS[side] or ''))
+		:node(OpponentDisplay)
+
+	return self
+end
+
+function UpperRow:addClass(class)
+	self.root:addClass(class)
+	return self
+end
+
+function UpperRow:versus(versus)
+	self.versusDisplay = mw.html.create('td')
+		:addClass('versus')
+		:node(versus)
+	return self
+end
+
+function UpperRow:winner(winner)
+	self.winnerValue = winner
+	return self
+end
+
+function UpperRow:create()
+	if self.winnerValue and self[self.winnerValue] then
+		self[self.winnerValue]:css('font-weight', 'bold')
+	end
+
+	return self.root
+		:node(self[_LEFT_SIDE])
+		:node(self.versusDisplay)
+		:node(self[_RIGHT_SIDE])
+end
+
+local Versus = Class.new(
+	function(self)
+		self.root = mw.html.create('div')
+		self.text = 'vs.'
+	end
+)
+
+function Versus:bestOf(bestOf)
+	if String.isNotEmpty(bestOf) then
+		self.bestOfDisplay = mw.html.create('abbr')
+			:attr('title', 'Best of ' .. bestOf)
+			:wikitext('Bo' .. bestOf)
+	end
+	return self
+end
+
+function Versus:score(matchData)
+	local leftScore, leftScore2, hasScore2, rightScore, rightScore2
+	leftScore, leftScore2, hasScore2 = HelperFunctions.getOpponentScore(
+		matchData.match2opponents[1],
+		matchData.winner == _WINNER_LEFT
+	)
+	rightScore, rightScore2, hasScore2 = HelperFunctions.getOpponentScore(
+		matchData.match2opponents[2],
+		matchData.winner == _WINNER_RIGHT,
+		hasScore2
+	)
+	self.text = leftScore .. ':' .. rightScore
+
+	if hasScore2 then
+		self.score2 = leftScore2 .. ':' .. rightScore2
+	end
+
+	return self
+end
+
+function Versus:create()
+	local lowerText, upperText
+	if self.score2 then
+		upperText = self.score2
+		lowerText = self.text
+	else
+		upperText = self.text
+		lowerText = self.bestOfDisplay
+	end
+	if lowerText then
+		return self.root
+			:node(mw.html.create('div')
+				:css('line-height', '1.1')
+				:node(upperText)
+			)
+			:node(mw.html.create('div')
+				:addClass('versus-lower')
+				:wikitext('(')
+				:node(lowerText)
+				:wikitext(')')
+			)
+	end
+	return self.root:wikitext(self.text)
+end
+
+local LowerRow = Class.new(
```

Comment:
```suggestion
local Details = Class.new(
```

Reply:
will adjust later today (have to adjust some other sandboxes that rely on the classes too)